### DISCUSSION TITLE
What happens when DataWriter::write() returned?

### DISCUSSION THREAD

**Discussion by ruoruoniao** [eProsima/Fast-DDS#3427] [Q&A]
I found that sometimes since DataWriter::write() returned, DataReader didn't receive published data.
But write() will use a little long time.
What is it waiting for?
What does it mean when it returns?

QoS:
    both RELIABLE_RELIABILITY_QOS
    both TRANSIENT_LOCAL_DURABILITY_QOS
    both max_blocking_time = {10, 0}

**Comment by jsan-rt** [comment]
Hi @ruoruoniao, 

We would need some more information regarding your application Qos configuration to know exactly why it's happening. Since you mentioned you are using `RELIABLE_RELIABILITY_QOS`, what you are seeing may be caused by reaching the resource limits specified in the [ResourceLimitsQosPolicy](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/standardQosPolicies.html?highlight=max_samples#resourcelimitsqospolicy) as explained [in this section of the documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/publisher/dataWriter/publishingData.html?highlight=max_blocking_time#blocking-of-the-write-operation). If this is the case the DataWriter is waiting for resources to become available and, if `max_blocking_time` is reached, it will return with a `TIMEOUT` return code.

**Reply by ruoruoniao** [reply]
Thanks for your reply~ 
All Qos settings ↓.
![image](https://user-images.githubusercontent.com/37649564/229665850-8ff9bcce-c796-4da3-8264-e1bc67f17f20.png)
Writer is the same as Reader.

So it just means send complete or timeout instead of a listener receipt?
